homeassistant/helpers/storage.py

```diff
@@ -187,7 +187,9 @@ def _write_data(self, path: str, data: Dict):
             os.makedirs(os.path.dirname(path))
 
         _LOGGER.debug('Writing data for %s', self.key)
-        json.save_json(path, data, self._private)
+        tmp_path = path + "__TEMP__"
```

Comment:
Mark file as not visible with `.`, and we should use a context manager to cleanup the temporary file after this.

Reply:
No need to clean up since we're using os.replace